src/Mollie/WC/Helper/PaymentMethodIconUrl.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+use Mollie\Api\Types\PaymentMethod;
+
+class Mollie_WC_Helper_PaymentMethodIconUrl
+{
+    /**
+     * @var null
+     */
+    private static $instance = null;
+    /**
+     * @var array
+     */
+    private $paymentMethodImages;
+
+    /**
+     * PaymentMethodIconUrl constructor.
+     * @param array $paymentMethodImages
+     */
+    private function __construct(array $paymentMethodImages)
+    {
+        $this->paymentMethodImages = $paymentMethodImages;
+    }
+
+    /**
+     * @param $paymentMethodImages
+     * @return Mollie_WC_Helper_PaymentMethodIconUrl|null
+     */
+    public static function getInstance($paymentMethodImages)
+    {
+        if (self::$instance == null)
+        {
+            self::$instance = new Mollie_WC_Helper_PaymentMethodIconUrl($paymentMethodImages);
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * @param string $paymentMethodName
+     * @return mixed
+     */
+    public function svgUrlForPaymentMethod($paymentMethodName)
+    {
+        return $this->paymentMethodImages? $this->paymentMethodImages[$paymentMethodName]['image']->svg : $this->fallToAssets($paymentMethodName);
+    }
+
+    /**
+     * @param string $paymentMethodName
+     * @return string
+     */
+    protected function fallToAssets($paymentMethodName)
+    {
+        if ( $paymentMethodName == PaymentMethod::CREDITCARD  && !is_admin()) {
+            return Mollie_WC_Plugin::getPluginUrl('assets/images/' . $paymentMethodName . 's.svg');
+        }
+        return Mollie_WC_Plugin::getPluginUrl('assets/images/' . $paymentMethodName . '.svg');
+    }
```

Comment:
Consider to make it more DRY also always do strict comparison.

Also, the name of the method could express better what it does. 
Consider to reflect the context of the body into the name of the method.

See example below.

```
protected function fallToAssets($paymentMethodName)
{
        $assetPath = "assets/images/{$paymentMethodName}.svg"

        if ( $paymentMethodName === PaymentMethod::CREDITCARD  && !is_admin()) {
            $assetPath = "assets/images/{$paymentMethodName}s.svg"
        }

        return $assetPath;
}
```